src/main/java/org/lisoft/lsml/model/item/AmmoWeapon.java

```diff
@@ -81,19 +92,43 @@ public Ammunition getAmmoType() {
         return ammoType;
     }
 
+    public int getVolleySize() {
+        return volleySize;
+    }
+    
+    public int getAmmoPerShot() {
+        return ammoPerShot;
```

Comment:
This will cause the total damage potential calculation to be wrong for C-UACs, I'll explain why, but first: I should document this better but in order to make it easier to reason about, LSML uses the following nomenclature:

* Shot: One trigger pull (for MGs/Flamers/RACs/etc this is one firing period worth of rounds).
* Round: A shot produces 1 or more rounds.
* Projectile: A round has 1 or more projectiles (or in the case of lasers, beams).
* Firing Period: Time between the start time of two consequent shots.

So a shot has rounds and rounds have projectiles. So the damage by one shot is: `damage / projectile * projectiles / rounds * rounds/shot` == `damage/shot`. And the damage per second is: `(damage / shot) / (seconds / shot)` == `damage / second`. Total damage potential is `damage = shots * damage / shot = total_rounds_of_ammo / (rounds/shot) * damage / shot`.

For example:
LB-10/X: 1 round per shot, 10 projectiles per shot, 1 damage per projectile. 10 damage per shot. 1 ammo (round) consumed per shot.
C-UAC/5: 2 rounds per shot, 1 projectile per shot, 2.5 damage per projectile. 5 damage per shot. 2 ammo (rounds) consumed per shot.
ER PPC: 1 round per shot, 1 projectile per shot, 10 damage per projectile. 10 damage per shot. 0 ammo consumed per shot.

This differs from the nomenclature in the data files which use `numFiring`, `damage`, `ammoPerShot`, `numPerShot`. Because the names in the data file weren't clear on how they were used, I decided on the current naming of these in LSML as it worked for all weapons.

Now, here's the tricky part, MWO isn't consistent in the use of the word "shot". Nor the meaning between different weapon types. For example, IS-LRM 20 has `numFiring=20` and `ammoPerShot=20` which aligns with the LSML meaning of a "shot". But if you look at for example C-UAC/5 it has `numFiring=2` and `ammoPerShot=1`... If you got into the game and load up a C-UAC/5 and shoot once, you'll consume 2 rounds of ammo. So the `ammoPerShot=1` value is not doing what you'd expect... And you can't just take `numFiring * ammoPerShot` because that's wrong for LRMs.

I have yet to find a way to parse the `ammoPerShot` field from the game files in a way that is consistent between all ammo weapons and that actually results in the correct amount of ammo consumed per (LSML) shot. However it turns out that `numFiring` has historically always lined up with the actual number of ammo rounds consumed per shot (as opposed to what `ammoPerShot` actually says), this is why LSML used `numFiring` for `getRoundsPerShot()` which was also used to mean `getAmmoPerShot()` because 1 round (of ammo) = 1 (round of) ammo. In LSML the amount of ammo rounds consumed per shot must be consistent with the LSML meaning of "shot" for the total damage capacity calculations in the info panel to be correct.

I had forgotten about this particular peculiarity of the data files, sorry for not catching this eariler. But you see, this kind of schenanigans is why LSML stays internally consistent rather than emulating whatever PGI does in the data files 😂.

Reply:
I know.  I had set it in one spot (just to capture the value) but then started to run into all of the above.  So I felt it was safer to just read it as PGI wrote it... and then LSML would set it appropriately later.  I'll change the creation set ammopershot using numFiring at initiation to be consistent in LSML and with how it actually acts in the game. I dont think getAmmoPerShot is called anywhere (you used getRoundsPerShot for that).  but I feel it would be more robust for future PGI shenanigans if LSML was ready for it.  

I ran into the exact same issues with how they handle the firing delay in ammo weapons.  PGI's inconsistent use of numFiring vs rounds per shot vs volleySize attributes is frustrating.  this could have easily been consistent for all ammo using weapons.

Reply:
I use getRoundsPer shot to determine the firing delay.  I think that is the correct reason to use it.  But, I'll search around and try to implement ammoPerShot in the right places.  initially, I did not want to mess with what was working for ammunition consumption too much, but the more I have fiddled with firing delay the more I have wanted to go look at ammunition consumption.

Reply:
ok, ran into a snag on using getAmmoPerShot. in WeaponSummary we shoud be using ammoPerShot.  but ammoPerShot only exists in the AmmoWeapon Class.  I am not certain how to properly check and then cast the generic weapon from the list to AmmoWeapon without messing up how lasers calculate damage over time. (which in the display should be infinity).

Reply:
I would prefer not to undo what I have put in place for this so far... but defer the final use (in the ammo use calculation) of this to a point where I know how to do this right>?

Reply:
btw the rest of this is done.

Reply:
I commented in WeaponSummary.